src/dao/migrations/20170909165723_create-repository-table.js

```diff
@@ -0,0 +1,25 @@
+const logger = require('~/src/logging').logger(module)
+const Repository = require('~/src/models/github/Repository')
+
+const REPOSITORY_TABLE_NAME = Repository.typeName
+
+exports.up = function (knex, Promise) {
+  logger.info('Attempting to run "up" on "create-repository-table"')
+  logger.info('Attempting to create repository table with table name: ', REPOSITORY_TABLE_NAME)
+
+  return knex
+    .schema
+    .createTableIfNotExists(REPOSITORY_TABLE_NAME, (table) => {
```

Comment:
I think this is a good start, but it might be nice if we included more information so that we could do some statistics on the data later on. I think a few more that would be nice to have are:

- `private: Boolean` # Whether this repo is private or public (this is subject to change...can we detect that?)
- `topics: String[]` # Topics associated with this repo e.g. dom, vdom, front-end, javascript
-  `created_at: Date` # The original data when the repo was created (I think)
- `language: String` # Main programming language of the repo
- `stargazers_count: Int` # The number of repo stars at the time the integration was enabled

Is this supposed to serve as a generic table for all source control providers? If so, some of the above _may_ not be applicable to all, but I think it's still worth having for the ones that do support it.

Some additional info regarding the data from GitHub: https://developer.github.com/v3/repos/#list-organization-repositories

Reply:
This table is merely meant for the installation:repository (1:many) relationship. Metadata about a repository should be in a different table. The primary key (id) can also be a foreign key into that repository metadata table since it's a 1:1 relationship. If we go with a setup like that, it might be better to call this table something like `InstallationRepositories` (or snakecased to be consistent)